components/dashboard/src/data/workspaces/fetchers.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { GitpodServer, WorkspaceInfo } from "@gitpod/gitpod-protocol";
+import { useCallback } from "react";
+import { useFeatureFlags } from "../../contexts/FeatureFlagContext";
+import { workspacesService } from "../../service/public-api";
+import { getGitpodService } from "../../service/service";
+
+type UseWorkspacesFetcherArgs = {
+    limit: number;
+};
+
+export type WorkspacesFetcherResult = WorkspaceInfo[];
+
+export const useWorkspacesFetcher = ({ limit = 50 }: UseWorkspacesFetcherArgs) => {
+    return useCallback(async (): Promise<WorkspacesFetcherResult> => {
+        // TODO: Can we update the backend api to sort & rank pinned over non-pinned for us?
+        const [infos, pinned] = await Promise.all([
+            getGitpodService().server.getWorkspaces({
+                limit,
+                includeWithoutProject: true,
+            }),
+            // Additional fetch for pinned workspaces
+            // see also: https://github.com/gitpod-io/gitpod/issues/4488
+            getGitpodService().server.getWorkspaces({
+                limit,
+                pinnedOnly: true,
+                includeWithoutProject: true,
+            }),
+        ]);
+
+        // Merge both data sets into one unique (by ws id) array
+        const workspacesMap = new Map(infos.map((ws) => [ws.workspace.id, ws]));
+        const pinnedWorkspacesMap = new Map(pinned.map((ws) => [ws.workspace.id, ws]));
+        const workspaces = Array.from(new Map([...workspacesMap, ...pinnedWorkspacesMap]).values());
+
+        return workspaces;
+    }, [limit]);
+};
+
+type UpdateWorkspaceDescriptionFetcherArgs = {
+    workspaceId: string;
+    newDescription: string;
+};
+export const useUpdateWorkspaceDescriptionFetcher = () => {
+    return useCallback(async ({ workspaceId, newDescription }: UpdateWorkspaceDescriptionFetcherArgs) => {
+        return await getGitpodService().server.setWorkspaceDescription(workspaceId, newDescription);
+    }, []);
+};
+
+type DeleteWorkspaceFetcherArgs = {
+    workspaceId: string;
+};
+export const useDeleteWorkspaceFetcher = () => {
+    const { usePublicApiWorkspacesService } = useFeatureFlags();
```

Comment:
I saw there was a PR that removed some uses of the flag, but wasn't sure if I could do the same here for some of these workspace apis.

Reply:
It can only be removed once it got successfully enabled for everyone. Currently, it is only active in staging (see configcat)

Reply:
Ah gotcha - I was confused from [this PR](https://github.com/gitpod-io/gitpod/pull/15712) - but realize it's discrete flags for different services of the public api, and not just one flag for the whole thing. Thanks.